src/consensus/vote.cpp

```diff
@@ -95,102 +100,92 @@ VoteManager::VoteManager(addr_t node_addr, std::shared_ptr<DbStorage> db, std::s
 
 VoteManager::~VoteManager() { daemon_->join(); }
 
-void VoteManager::setNetwork(std::weak_ptr<Network> network) { network_ = move(network); }
+void VoteManager::setNetwork(std::weak_ptr<Network> network) { network_ = std::move(network); }
 
 void VoteManager::retreieveVotes_() {
+  LOG(log_si_) << "Retrieve unverified votes from DB";
   auto unverified_votes = db_->getUnverifiedVotes();
   for (auto const& v : unverified_votes) {
-    auto pbft_round = v.getRound();
-    auto hash = v.getHash();
+    auto pbft_round = v->getRound();
+    auto hash = v->getHash();
     {
       uniqueLock_ lock(unverified_votes_access_);
       if (unverified_votes_.count(pbft_round)) {
         unverified_votes_[pbft_round][hash] = v;
       } else {
-        std::unordered_map<vote_hash_t, Vote> votes{std::make_pair(hash, v)};
-        unverified_votes_[pbft_round] = votes;
+        std::unordered_map<vote_hash_t, std::shared_ptr<Vote>> votes{std::make_pair(hash, v)};
+        unverified_votes_[pbft_round] = std::move(votes);
       }
     }
-    LOG(log_dg_) << "Retrieved unverified vote " << v;
+    LOG(log_dg_) << "Retrieved unverified vote " << *v;
   }
 
+  LOG(log_si_) << "Retrieve verified votes from DB";
   auto verified_votes = db_->getVerifiedVotes();
   for (auto const& v : verified_votes) {
-    auto pbft_round = v.getRound();
-    auto hash = v.getHash();
-
     // Rebroadcast our own next votes in case we were partitioned...
-    if (v.getVoterAddr() == node_addr_ && v.getStep() >= FIRST_FINISH_STEP &&
+    if (v->getVoterAddr() == node_addr_ && v->getStep() >= FIRST_FINISH_STEP &&
         db_->getPbftMgrField(PbftMgrRoundStep::PbftStep) > EXTENDED_PARTITION_STEPS) {
-      vector<Vote> votes = {v};
+      std::vector<std::shared_ptr<Vote>> votes = {v};
       if (auto net = network_.lock()) {
-        net->onNewPbftVotes(move(votes));
+        net->onNewPbftVotes(std::move(votes));
```

Comment:
Why is this called inside for loop and not after that loop is done ?

Reply:
When nodes get partition, that don't need all verified votes. Partitioned nodes will only need 2t+1 next votes in one step  to jump to the correct PBFT round. Sending them one by one will be faster for partitioned nodes jump to correct PBFT round.
This need to improve later for sending votes in bundle: https://github.com/Taraxa-project/taraxa-node/issues/1158